File: cattle_management_system/tmp_replace.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    idx = 0
    while idx < len(content):
        # Allow multi-line whitespace around the text
        match = re.search(r'ScaffoldMessenger\.of\(\s*([a-zA-Z0-9_]+)\s*\)(?:\.\.clearSnackBars\(\))?\.showSnackBar\(', content[idx:])
        if not match:
            break
        
        start_idx = idx + match.start()
        context_var = match.group(1)
        
        paren_count = 1
        curr_idx = start_idx + len(match.group(0))
        snack_bar_inner = ""
        while curr_idx < len(content) and paren_count > 0:
            char = content[curr_idx]
            if char == '(':
                paren_count += 1
            elif char == ')':
                paren_count -= 1
            if paren_count > 0:
                snack_bar_inner += char
            curr_idx += 1

        if curr_idx < len(content) and content[curr_idx] == ';':
            curr_idx += 1
        
        # Now parse the text content
        # It's usually SnackBar(content: Text('message'))
        # But we only have inside showSnackBar(...)
        text_match = re.search(r'content:\s*Text\((.*?)\)', snack_bar_inner, re.DOTALL)
        if text_match:
            text_inner = text_match.group(1).strip()
            
            is_error = False
            lower_inner = text_inner.lower()
            if 'colors.red' in snack_bar_inner.lower() or 'error' in lower_inner or 'fail' in lower_inner or 'exception' in lower_inner or text_inner.startswith('e.') or '$e' in text_inner:
                is_error = True

            if 'required' in lower_inner or 'not found' in lower_inner:
                is_error = True
            
            if 'success' in lower_inner or 'updated' in lower_inner or 'created' in lower_inner or 'added' in lower_inner or 'deleted' in lower_inner:
                if 'fail' not in lower_inner and 'error' not in lower_inner:
                    is_error = False
            
            # fallback to error if variable is `e`
            if text_inner == 'e':
                is_error = True

            method = "showError" if is_error else "showSuccess"
            print(f"[{os.path.basename(filepath)}] Found: {text_inner} => {method}")

        idx = curr_idx

def main():
    lib_dir = "d:\\Desktop\\Cattle-Management-System\\cattle_management_system\\lib"
    for root, dirs, files in os.walk(lib_dir):
        for file in files:
            if file.endswith('.dart'):
                process_file(os.path.join(root, file))

if __name__ == "__main__":
    main()

File: cattle_management_system/tmp_replace.dart
import 'dart:io';

void main() async {
  final dir = Directory('lib');
  final files = dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));

  int totalReplacements = 0;

  for (var file in files) {
    if (file.path.contains('app_feedback') || file.path.contains('main.dart')) continue;

    String content = await file.readAsString();
    bool changed = false;

    // A simpler replacement strat:
    // Regex to match the whole scaffolding call
    final exp = RegExp(r'ScaffoldMessenger\.of\(\s*([a-zA-Z0-9_]+)\s*\)(?:\.\.clearSnackBars\(\))?\.showSnackBar\(\s*SnackBar\(\s*content:\s*Text\(([^)]+)\)\s*(?:,\s*backgroundColor:\s*[^)]+)?\s*\)\s*(?:,)?\s*\);?', multiLine: true, dotAll: true);

    String newContent = content.replaceAllMapped(exp, (match) {
      String ctxVar = match.group(1)!;
      String textInner = match.group(2)!.trim();
      
      String lowerInner = textInner.toLowerCase();
      
      bool isError = false;
      String matchStr = match.group(0)!.toLowerCase();
      if (matchStr.contains('colors.red') ||
          lowerInner.contains('error') || lowerInner.contains('fail') || lowerInner.contains('exception') ||
          textInner.startsWith('e.') || textInner.contains(r'$e') || textInner == 'e') {
        isError = true;
      }
      
      if (lowerInner.contains('required') || lowerInner.contains('not found')) {
        isError = true;
      }
      
      if (lowerInner.contains('success') || lowerInner.contains('updated') || lowerInner.contains('created') || lowerInner.contains('added') || lowerInner.contains('deleted')) {
        if (!lowerInner.contains('fail') && !lowerInner.contains('error')) {
          isError = false;
        }
      }
      
      String method = isError ? 'showError' : 'showSuccess';
      print('[${file.path}] Repl: $textInner => $method');
      changed = true;
      totalReplacements++;
      return 'AppFeedback.$method($ctxVar, $textInner);';
    });
    
    // There might be some calls without semicolon or with different formatting.
    final exp2 = RegExp(r'ScaffoldMessenger\.of\(\s*([a-zA-Z0-9_]+)\s*\)\.showSnackBar\(([^;]+)\);?', multiLine: true, dotAll: true);
    
    newContent = newContent.replaceAllMapped(exp2, (match) {
      String inner = match.group(2)!;
      if (inner.contains('SnackBar(content: Text(') && !inner.contains('AppFeedback')) {
         // fallback manually
         final textMatch = RegExp(r'Text\((.*?)\)', dotAll: true).firstMatch(inner);
         if (textMatch != null) {
            String textInner = textMatch.group(1)!.trim();
            String method = textInner.toLowerCase().contains('error') ? 'showError' : 'showSuccess';
            print('[${file.path}] fallback Repl: $textInner => $method');
            changed = true;
            totalReplacements++;
            return 'AppFeedback.$method(${match.group(1)!}, $textInner);';
         }
      }
      return match.group(0)!; // no change
    });


    if (changed) {
      if (!newContent.contains('import \'package:cattle_management_system/core/utils/app_feedback.dart\';') &&
          !newContent.contains('import \'../../core/utils/app_feedback.dart\';')) {
        
        // Find flutter material import or first import
        int importIdx = newContent.lastIndexOf(RegExp(r"import\s+'[^']+';"));
        if (importIdx != -1) {
           int endOfImport = newContent.indexOf('\n', importIdx);
           newContent = newContent.substring(0, endOfImport + 1) + 
                        "import 'package:cattle_management_system/core/utils/app_feedback.dart';\n" + 
                        newContent.substring(endOfImport + 1);
        } else {
           newContent = "import 'package:cattle_management_system/core/utils/app_feedback.dart';\n" + newContent;
        }
      }
      await file.writeAsString(newContent);
    }
  }
  print("Total replacements: $totalReplacements");
}
